### PR TITLE
Handling Duplicate Participants and Form Views

### LINKED ISSUE #90: Magical Modal for Searching Users
@vjoseph2 I heard you had this module that  was used in Scott's attendance section. It allows you to pick the right user after a search? Can you point me in the direction of this/do you have a test server so I can see it? This is important since it will not only be used in attendance but anytime information is being inserted about a participant.  Like for intake packets, referral packets, etc. 
- [x] Pass participant information as parameters in a sql query to find matches
- [x]  Grab all returned results and display in modal
- [x] Allow user to select a matching participant
- [x] Store the selected participant ID for use in referral stored proceedures

### LINKED ISSUE #165: Edit forms on View Participant page
Since we are going to be implementing edit functionality for all of our forms, could you guys create view/edit buttons for each of the forms on the View Participant page?

### LINKED ISSUE #179: What should we display if a participant has more than one form(s) completed?
This is a question I have for the edit/update form functionality that we are implementing.

What should we display on the edit/update page?

### PR DESCRIPTION
**Pull request by vmjoseph**
- Adds ability to view users forms
- Allows blanks in names
- Adds ability to check database for duplicates and assign forms to duplicate participants rather than create a new participant
- Fixes #179
- Fixes #90 
- Fixes #165

### CODE REVIEW

File: views/forms/intake_packet.php [hunk 1]
```diff
@@ -326,7 +321,7 @@
             die(pg_result_error($formID));
             $_SESSION['form-error'] = true;
             $_SESSION['error-state'] = $state;
-            header("Location: /form-success");
+            //header("Location: /form-success");
```
Comment (michelleburd) [inline views/forms/intake_packet.php]: Uncomment these

File: views/forms/intake_packet_view.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (michelleburd) [inline views/forms/intake_packet_view.php]: I think we can remove this file

File: views/forms/referral_form.php [hunk 3]
```diff
@@ -1013,7 +1007,8 @@
             </div>  <!-- panel group end -->
             <br>
 
-            <button id="btnRegister" onclick="submitAll()" class="cpca btn controls" style="margin-bottom: 20px;">Submit</button>
+			<?php include('form_duplicate_check.php')?>
+            <!--<button id="btnRegister" onclick="submitAll()" class="cpca btn controls" style="margin-bottom: 20px;">Submit</button>-->
```
Comment (michelleburd) [inline views/forms/referral_form.php]: Uncomment?

File: views/forms/self_referral_form.php [hunk 4]
```diff
@@ -413,8 +408,8 @@ classAssigned := $18::TEXT,
                 </form>
             </div>  <!-- panel group end -->
             <br>
-
-            <button id="btnRegister" onclick="submitAllSelf()" class="cpca btn controls" style="margin-bottom: 20px;">Submit</button>
+            <?php include('form_duplicate_check.php')?>
+            <!--<button id="btnRegister" onclick="submitAllSelf()" class="cpca btn controls" style="margin-bottom: 20px;">Submit</button>-->
```
Comment (michelleburd) [inline views/forms/self_referral_form.php]: Uncomment?

File: public/js/duplicateCheck.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	$(".state").text();
+	$(".aptt").text();
+
+    // Sets the left side of the omdal to hold a fixed view of entered information
+    $(".followNameHolder").html("<p><b>First Name:</b> "+firstNameValue+"</p>"+
+                        "<p><b>Middle Name:</b> "+middleinit+"</p>"+
+                        "<p><b>Last Name:</b> "+lastname+"</p>"+
+                        "<p><b>DOB:</b> "+dob+"</p>"+
+                        "<p><b>Race:</b> "+(race == null ? " ": race)+"</p>"+
+                        "<p><b>Sex:</b> "+middleinit+"</p>"+
```
Comment (ghost) [inline public/js/duplicateCheck.js]: Sex shouldn't have the middle initial.

File: public/js/duplicateCheck.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                        "<p><b>Address:</b> "+address + " "+apt +"</p>"+
+                        "<p><b>City:</b> "+city+"</p>"+
+                        "<p><b>State:</b> "+(state == null ? " ": state)+"</p>"+
+                        "<p><b>Zip:</b> "+zip+"</p>");
+
+	$.ajax({
```
Comment (ghost) [inline public/js/duplicateCheck.js]: Whitespace in this file is kind of crazy. Hard to follow scope.

File: views/forms/forms_view.php [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+global $db, $params, $route;
+$peopleid = $params[0];
+
+# Get participant first name, last name, middle
+$result = $db->query("SELECT firstname, lastname, middleinit FROM people WHERE peopleid = $1", [$peopleid]);
+$name = pg_fetch_assoc($result);
+extract($name);
+
+# Displays the participant's name in the title according to whether they have a middle name
+$route['title'] = "Forms Completed for " . ucwords($firstname . " " . ($middleinit ?: "") . " " . $lastname);
```
Comment (ghost) [inline views/forms/forms_view.php:23]: Doesn't this give them `""` if they do actually have a middleInit? 
```
$v = 'My Value';
$r = ($v) ?: 'No Value'; // $r is set to 'My Value' because $v is evaluated to TRUE
```

Comment (blackjack26) [reply views/forms/forms_view.php:23]: I'm pretty sure this works. Also you're example explains why it works I believe

File: page_permissions.ini [hunk 8]
```diff
@@ -61,9 +61,15 @@
 
 ; intake / referral
 /referral-form = ALLOW User
+/referral-form/*/*/* = ALLOW User
```
Comment (blackjack26) [inline page_permissions.ini]: This could just be `/referral-form/**` which will get any number of route params 1 or more

File: page_permissions.ini [hunk 9]
```diff
@@ -61,9 +61,15 @@
...
 /form-success = ALLOW User
 /self-referral-form = ALLOW User
+/self-referral-form/*/*/* = ALLOW User
```
Comment (blackjack26) [inline page_permissions.ini]: `/self-referral-form/**`

File: page_permissions.ini [hunk 10]
```diff
@@ -61,9 +61,15 @@
...
 /intake-packet = ALLOW User
+/intake-packet/*/*/* = ALLOW User
```
Comment (blackjack26) [inline page_permissions.ini]: `/intake-packet/**`

File: public/js/duplicateCheck.js [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+* @copyright 2017 Marist College
+* @version 1.1
+* @since 1.1
+*
+*
+*/
+
+$(document).ready(function(){
+
+var pathname = window.location.pathname; // Returns path only
```
Comment (blackjack26) [inline public/js/duplicateCheck.js]: Indent this section of code since it is within the `ready()` function

File: public/js/duplicateCheck.js [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+var state ;
+var dob ;
+var race;
+var sex;
+
+/**
+* Checking the page route to see what page the
+* user is on; this will tell the modal what js function
+* to run depending on the form that is being completed
+* @params pathN string that contains the path route
```
Comment (blackjack26) [inline public/js/duplicateCheck.js]: `@param` instead of `@params`

File: public/js/duplicateCheck.js [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+*/  
+function routePage(pathN){
+    if(pathN == "/referral-form"){
+        submitAll();
+    }else if ( pathN == "/self-referral-form"){
+        submitAllSelf()
+    }else if( pathN == "/intake-packet"){
```
Comment (blackjack26) [inline public/js/duplicateCheck.js]: I would suggest sticking to one style of spacing.

Either (line 49):
```php
}else if ( pathN == "/self-referral-form"){
```
or (line 51):
```php
}else if( pathN == "/intake-packet"){
```

Personally I'd suggest to at least put a space before the `else` keyword.

File: views/agency-requests/agency_requests_results.php [hunk 14]
```diff
@@ -73,11 +74,27 @@ function checkEmpty($rowValue){
...
 
 				<ul class="list-group sublist">
-						<?= "<b>DOB: </b>".(isset($row['dateofbirth']) ? checkEmpty($row['dateofbirth']) : '').
-                        " | "." <b>Program Start Date: </b> ".
-                        (isset($row['tentativestartdate']) ? checkEmpty($row['tentativestartdate']) : '').
-                        " | "."<b>Sex: </b> ".
-                        (isset($row['sex']) ? checkEmpty($row['sex']) : ''); ?>
+						<?php
+                        $formInfo = $db->query("SELECT  
+                        formid, selfreferralid, intakeinformationid, agencyreferralid,
```
Comment (blackjack26) [inline views/agency-requests/agency_requests_results.php]: Indent this section of code to show it's part of the above function

File: views/agency-requests/agency_requests_results.php [hunk 15]
```diff
@@ -73,11 +74,27 @@ function checkEmpty($rowValue){
...
+                        FROM forms
+                        INNER JOIN participants ON forms.participantid = participants.participantid
+                        LEFT JOIN selfreferral ON forms.formid = selfreferral.selfreferralid
+                        LEFT JOIN intakeinformation ON forms.formid = intakeinformation.intakeinformationid
+                        LEFT JOIN agencyreferral ON forms.formid = agencyreferral.agencyreferralid
+                        WHERE participants.participantid= $1",[$row['participantid']]);
+                        while($formResults  = pg_fetch_assoc($formInfo)){
+                        ?>   
+                         <div>
+                        <b>DOB: </b><?=checkEmpty($formResults['dateofbirth']);?> |
```
Comment (blackjack26) [inline views/agency-requests/agency_requests_results.php]: Indent HTML tags

File: views/forms/forms_view.php (hunk 7 above)
Comment (ghost) [reply views/forms/forms_view.php:23]: You're right. I spoke too quick =(

File: views/forms/form_duplicate_check.php [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+                    <i>Click to select a participant.</i>
+            </div>
+				<div class="nameCol">
+				</div>
+			</div>
+      </div>
+	  </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary cancel" data-dismiss="modal">Create New Participant</button>
+        <button type="button" class="btn cpca">Edit Form Information</button>
```
Comment (blackjack26) [inline views/forms/form_duplicate_check.php]: This button does nothing. Suggest a different name like "Cancel" or "Go Back".

File: views/forms/form_duplicate_check.php [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+            <div class="sticky-top" style="background-color:white;">
+                    <i>Click to select a participant.</i>
+            </div>
+				<div class="nameCol">
+				</div>
+			</div>
+      </div>
+	  </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary cancel" data-dismiss="modal">Create New Participant</button>
```
Comment (blackjack26) [inline views/forms/form_duplicate_check.php]: I'd name this "Yes, this is a new participant" since you pose the question "Is this a new participant?".

File: views/forms/form_match.php [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+?>
+ <ul class='list-group'>
+
+    <?php
+    // Display all matching values as a list item
+    while($row = pg_fetch_assoc($result)){
+    ?>
+         <li class="list-group-item " id="<?=$row['peopleid']?>">
```
Comment (blackjack26) [inline views/forms/form_match.php]: When hovering over these I can't tell I'm supposed to click on them. Maybe suggest `cursor: pointer`

File: views/forms/form_match.php [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+            <?= $row['lastname'].", ".$row['firstname']. " ". $row['middleinit'];?>
+
+            <?= " <b>DOB: </b>".$row['dateofbirth']." | ".
+            " <b>Address: </b> ".$row['addressnumber'].$row['aptinfo']." ".$row['street']. " ".$row['city']. " ".$row['state']. " ".$row['zipcode']." | ".
+            " <b>Phone: </b>".$row['phonenumber']." (".$row['phonetype'].")| ".
```
Comment (blackjack26) [inline views/forms/form_match.php]: When `$row['phonetype']` doesn't exist it just displays "()"

**Review by blackjack26 — CHANGES_REQUESTED** [review]

**Review by blackjack26 — APPROVED** [review]
